iglo/league/tests/test_egd_export.py

```diff
@@ -0,0 +1,466 @@
+        
+        # Create members with specific ranks
+        self.member1 = MemberFactory(group=self.group, player=self.player1, rank=2700)  # 7d
+        self.member2 = MemberFactory(group=self.group, player=self.player2, rank=2300)  # 3d
+        self.member3 = MemberFactory(group=self.group, player=self.player3, rank=1900)  # 2k
+        self.member4 = MemberFactory(group=self.group, player=self.player4, rank=1700)  # 4k
+        
+        # Round 1: One EGD-eligible game (player1 vs player2)
+        self.game1 = GameFactory(
+            group=self.group,
+            round=self.round1,
+            black=self.member1,
+            white=self.member2,
+            winner=self.member1,
+            win_type=WinType.RESIGN
+        )
+        
+        # Round 1: One non-EGD-eligible game (player3 vs player4)
+        # Player 3 doesn't have EGD approval
+        self.game2 = GameFactory(
+            group=self.group,
+            round=self.round1,
+            black=self.member3,
+            white=self.member4,
+            winner=self.member4,
+            win_type=WinType.RESIGN
+        )
+        
+        # Round 2: One EGD-eligible game (player2 vs player4)
+        self.game3 = GameFactory(
+            group=self.group,
+            round=self.round2,
+            black=self.member2,
+            white=self.member4,
+            winner=self.member2,
+            win_type=WinType.POINTS
+        )
+        
+        # Round 2: One BYE game (player1)
+        self.game4 = GameFactory(
+            group=self.group,
+            round=self.round2,
+            winner=self.member1,
+            win_type=WinType.BYE
+        )
+        
+        # Round 3: One unplayed game between EGD-eligible players
+        self.game5 = GameFactory(
+            group=self.group,
+            round=self.round3,
+            black=self.member1,
+            white=self.member4,
+            win_type=None  # Not played
+        )
+        
+        # Round 3: One NOT_PLAYED game (forfeit) between EGD-eligible players
+        self.game6 = GameFactory(
+            group=self.group,
+            round=self.round3,
+            black=self.member2,
+            white=self.member3,  # Player without EGD approval
+            win_type=WinType.NOT_PLAYED,
+            winner=self.member2  # Winner awarded without play
+        )
+        
+        # Round 4: No EGD-eligible games, only non-eligible pairs
+        self.game7 = GameFactory(
+            group=self.group,
+            round=self.round4,
+            black=self.member1,
+            white=self.member3,  # Player without EGD approval
+            winner=self.member1,
+            win_type=WinType.RESIGN
+        )
+        
+        self.game8 = GameFactory(
+            group=self.group,
+            round=self.round4,
+            black=self.member4,
+            white=self.member2,
+            win_type=None  # Not played
+        )
+
+    def test_complex_egd_export_with_edge_cases(self):
```

Comment:
instead of writing "with_edge_cases" I'd just write what edge case and test single edge case per test

Reply:
I forgot to uncomment the assertion here. 
Fixed.

But I'm not willing to work more on this test, its annoying. 
I can delete it, but it serves a value when running all tests.